assets/scripts/gx_cloud/experimental/onboarding_script.py

```diff
@@ -6,177 +7,117 @@
-assert (
-    path_to_validator_batch
-), "Please set path_to_validator_batch. This can be a local filepath or a remote URL."
+# Get or add Datasource
+datasource: BaseDatasource = context.sources.add_or_update_pandas(datasource_name)
 
-# Set variable for creating a Checkpoint
-checkpoint_name = None
-assert checkpoint_name, "Please set checkpoint_name."
+# Provide an Asset name
+asset_name = None
+assert asset_name, "Please set asset_name."
 
-# Set variable to get a Batch of data to validate against the new Checkpoint
-path_to_batch_to_validate = None  # e.g. "https://raw.githubusercontent.com/great-expectations/gx_tutorials/main/data/yellow_tripdata_sample_2019-01.csv"
+# Provide a path to data
+path_to_data = None
+# to use sample data uncomment next line
+# path_to_data = "https://raw.githubusercontent.com/great-expectations/gx_tutorials/main/data/yellow_tripdata_sample_2019-01.csv"
 assert (
-    path_to_batch_to_validate
-), "Please set path_to_batch_to_validate. This can be a local filepath or a remote URL."
+    path_to_data
+), "Please set path_to_data. This can be a local filepath or a remote URL."
 
-# Create Datasource
-# For simplicity, this script creates a Datasource with a PandasExecutionEngine and a RuntimeDataConnector
+# Get or add Asset
 try:
-    datasource: BaseDatasource = context.get_datasource(datasource_name=datasource_name)
-except ValueError:
-    datasource_yaml = f"""
-    name: {datasource_name}
-    class_name: Datasource
-    execution_engine:
-        class_name: PandasExecutionEngine
-    data_connectors:
-        {data_connector_name}:
-            class_name: RuntimeDataConnector
-            batch_identifiers:
-            - path
-    """
-    # Test your configuration:
-    datasource: BaseDatasource = context.test_yaml_config(datasource_yaml)
-
-    # Save your Datasource:
-    datasource: BaseDatasource = context.add_or_update_datasource(datasource=datasource)
+    asset: CSVAsset = datasource.get_asset(asset_name=asset_name)
+except LookupError:
+    asset: CSVAsset = datasource.add_csv_asset(
+        asset_name, filepath_or_buffer=path_to_data
+    )
 
-print(f"\n{20*'='}\nDatasource Config\n{20*'='}\n")
-pprint.pprint(datasource.config)
+# Build BatchRequest
+batch_request: BatchRequest = asset.build_batch_request()
 
-# Create a new Expectation Suite
-try:
-    expectation_suite: ExpectationSuite = context.get_expectation_suite(
-        expectation_suite_name=expectation_suite_name
-    )
-    expectation_suite_ge_cloud_id = expectation_suite.ge_cloud_id
-except StoreBackendError:
-    expectation_suite: ExpectationSuite = context.add_or_update_expectation_suite(
-        expectation_suite_name=expectation_suite_name
-    )
-    expectation_suite_ge_cloud_id = expectation_suite.ge_cloud_id
+print(f"\n{20*'='}\nDatasource Config\n{20*'='}\n")
+pprint.pprint(datasource.json())
```

Comment:
```suggestion
print(f"\n{20*'='}\nDatasource Config\n{20*'='}\n")
pprint.pprint(datasource.dict())
```

`.json()` returns an actual json string.
What you probably want is `.dict()`

https://docs.pydantic.dev/latest/usage/exporting_models/


---------


Unrelated to `.dict()` vs `.json()` ...
`pprint` also has a `pformat` function which just returns a pretty printed formatted string.
https://docs.python.org/3/library/pprint.html#pprint.pformat

So you can do things like this.
```python
from pprint import pformat as pf

...

print(f"\n{20*'='}\nDatasource Config\n{20*'='}\n{pf(datasource.dict())}")

```